File: 78-subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> temp;
    void solve(vector<int>& nums, int i){
        if(i >= nums.size()){
            ans.push_back(temp);
            return;
        }
        temp.push_back(nums[i]);
        solve(nums, i+1);
        temp.pop_back();
        solve(nums, i+1);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        solve(nums, 0);
        return ans;
    }
};

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        res = []
        for mask in range(1 << n):
            sub = []
            for i in range(n):
                if mask & (1 << i):
                    sub.append(nums[i])
            res.append(sub)
        return res
        